### PR TITLE
[510] 510 compilation.

### PR DESCRIPTION
**Pull request by tigercat2000**
Targeted at the `byond510` branch.

@Aurorablade Will need your help with the organ system to update the branch, once it's merged.

Changes so far:
 - Fullscreen HUD system redesigned, thanks to @neersighted (https://github.com/tgstation/-tg-station/pull/14981)
  - Fullscreen HUDs work properly in 510 and 509, as far as I can tell.
  - Fullscreen overlays (flashed, blindness, nearsightedness, etc) will now fade out, instead of abruptly popping out.
  - The nearsighted overlay gives you a clear 9x9 vision, and a 15x15 "dithered" vision. The tint from welding goggles will give you a 5x5 clear vision, 9x9 dithered vision, and 15x15 blind vision.
  - The old Bay `pain` overlay has been removed. This is **not** the brute damage overlay, the red ring you get when you take damage, that's still around. I'm not entirely sure what it *is* because as far as I can tell, it has never worked.
 - Flashing refactored (https://github.com/tgstation/-tg-station/pull/8286, https://github.com/tgstation/-tg-station/pull/10668, https://github.com/tgstation/-tg-station/pull/15217)
 - Bygex completely removed, as it is native in byond 510.
  - The `auto-convert .sav notes to DB notes` system has been ripped out, partly because I didn't know how to make it work with the new byond regex, partly because we have never used it.
  - text2list & list2text have been removed as well, as they are native in byond 510 (`splittext` and `jointext` respectively.
 - stddef.dm added to .gitignore, because it was annoying me trying to keep it from ending up in a commit while refering to it.
 - A few unique things to the `weakeyes` var have been removed, as they were getting in the way.
 - `eyecheck()` is now `check_eye_prot()`
 - Xenos no longer 100% override vision code. Still shit.
 - For *now*, xenomorphs cannot take eye damage. When the organ PR is done and merged into the 510 branch, they probably will be able to.

### CODE REVIEW

**Comment by TheDZD** [comment]
Need to update [these two lines](https://github.com/ParadiseSS13/Paradise/blob/master/.travis.yml#L10-L11).

**Comment by tigercat2000** [comment]
aahh knew i would forget something

**Comment by Spacemanspark** [comment]
Please make sure flashing doesn't permastun cyborgs with this change.
I don't know what caused it/ if /tg/ fixed it, but they have/ had this horrid bug where you would get permanently stunned if you got flashed.

**Comment by tigercat2000** [comment]
@Spacemanspark They, at the very least, definitely properly work with stationary flashers. I did grab all the fixes I could find for the flash code. I'll test normal flashes, but i'm sure it's fine.

**Comment by tigercat2000** [comment]
update: borgs don't get flashed at all rip

**Comment by Spacemanspark** [comment]
The irony.

**Comment by Fox-McCloud** [comment]
awww, no thrown alerts yet? =p

**Comment by Tastyfish** [comment]
I'll throw in my PS changes as soon as this gets merged.

File: code/modules/mob/living/carbon/alien/life.dm [hunk 1]
```diff
@@ -36,4 +36,24 @@
 			src << "<span class='danger'>You feel a searing heat in your lungs!</span>"
 		fire_alert = max(fire_alert, 1)
 	else
-		fire_alert = 0
\ No newline at end of file
+		fire_alert = 0
+
+/mob/living/carbon/alien/update_sight()
+	if(stat == 2 || (XRAY in mutations))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/alien/life.dm]: magical numbers :angry:

Comment (tigercat2000) [inline code/modules/mob/living/carbon/alien/life.dm]: It's alien code, there's a whole lot worse.

**Pull request merged**